Hunting Queries/MultipleDataSources/MultiplePasswordResetbyUser.txt

```diff
@@ -0,0 +1,61 @@
+// Name: Multiple Password Reset by user
+// Description: This query will determine multiple password resets by user across multiple data sources. 
+// Account manipulation including password reset may aid adversaries in maintaining access to credentials 
+// and certain permission levels within an environment.
+// 
+// Id: 0b9ae89d-8cad-461c-808f-0494f70ad5c4
+//
+// DataSource: #AuditLogs, #Syslog, #SecurityEvent, #OfficeActivity
+//
+// Tactics: #Discovery, #CredentialAccess
+//
+let timeframe = 1d;
+let PerUserThreshold = 5;
+let TotalThreshold = 100;
+let PasswordResetMultiDataSource =
+(union isfuzzy=truegit
+(
+//WEF Password reset events
+//4723: An attempt was made to change an account's password
+//4724: An attempt was made to reset an accounts password
+SecurityEvent
+| where TimeGenerated >= ago(timeframe)
+| where EventID in ("4723","4724")
+| extend DataSource="WEF"
+| project DataSource, TimeGenerated,AccountType,Account
+),
+(
+//Azure Active Directory Password reset events
+AuditLogs
+| where Category == "UserManagement" and Result == "success"
+| where TimeGenerated >= ago(timeframe)
+| where OperationName in ("Change user password","Reset user password","Change password (self-service)","Reset password (self-service)")
+| extend DataSource="Azure Active Directory"
+| extend AccountType = tostring(TargetResources[0].type, Account = tostring(TargetResources[0].userPrincipalName))
+| project DataSource, TimeGenerated,AccountType,Account,Computer
+),
+(
+//OfficeActive ActiveDirectory Password reset events
+OfficeActivity
+| where OfficeWorkload == "AzureActiveDirectory" 
+| where TimeGenerated >= ago(timeframe)
+| where Operation in ("Change user password.","Reset user password.") and ResultStatus =="Success"
+| extend DataSource="OfficeActivity", AccountType = UserType, Account = OfficeObjectId 
+| project DataSource, TimeGenerated,AccountType,Account
+),
+(
+// Unix syslog password reset events
+Syslog
+| where TimeGenerated >= ago(timeframe)
+| where Facility in ("auth","authpriv")
+| where SyslogMessage contains "password changed for"
+| extend AccountType = iif(SyslogMessage contains "root", "Root", "Non-Root"), DataSource="Linux"
+| parse Syslogmessage with * "password changed for" Account
+| project DataSource, TimeGenerated, AccountType, Account 
+)
+);
+PasswordResetMultiDataSource 
+| project DataSource, TimeGenerated,AccountType,Account
+|fork
+PerUserReset = (summarize min(TimeGenerated), max(TimeGenerated), Total=count() by Account, DataSource | where Total > PerUserThreshold)
+TotalReset = (summarize min(TimeGenerated), max(TimeGenerated), Total=count() by DataSource | where Total > TotalThreshold)
```

Comment:
TotalReset does not come through...did you want to union it and lablel it or did you not want this to come thru?

Reply:
submitted new commit with changes